casper/src/test/scala/coop/rchain/casper/engine/RunningHandleBlockMessageSpec.scala

```diff
@@ -37,98 +37,98 @@ import scala.collection.concurrent.TrieMap
+//
+//  private def endpoint(port: Int): Endpoint = Endpoint("host", port, port)
+//  private def peerNode(name: String, port: Int): PeerNode =
+//    PeerNode(NodeIdentifier(name.getBytes), endpoint(port))
+//
+//  private def alwaysSuccess: PeerNode => Protocol => CommErr[Unit] = kp(kp(Right(())))
+//
+//  private def alwaysIgnoreF: BlockHash => Task[Running.IgnoreCasperMessageStatus] =
+//    kp(IgnoreCasperMessageStatus(true, BlockIsWaitingForCasper).pure[Task])
+//  private def alwaysDoNotIgnoreF: BlockHash => Task[Running.IgnoreCasperMessageStatus] =
+//    kp(IgnoreCasperMessageStatus(false, DoNotIgnore).pure[Task])
+//
+//  private def alwaysValidBlockF: (BlockMessage, PeerNode) => Task[Boolean] =
+//    kp2[BlockMessage, PeerNode, Task[Boolean]](true.pure[Task])
+//  private def alwaysBadBlockF: (BlockMessage, PeerNode) => Task[Boolean] =
+//    kp2[BlockMessage, PeerNode, Task[Boolean]](false.pure[Task])
+//
+//  private def casperAddF(bh: BlockHash): Task[Unit] = (sentToCasper += bh).pure[Task].void
+//  private val hashHandlerLock                       = TrieMap.empty[BlockHash, Semaphore[Task]]
+//
+//  override def beforeEach(): Unit = {
+//    transportLayer.reset
+//    transportLayer.setResponses(alwaysSuccess)
+//    sentToCasper.remove(validBlock.blockHash)
+//    blockStore.clear.runSyncUnsafe()
+//  }
+
+//  describe("Running") {
+//    describe("handleBlockMessage") {
+//      it("should pass good block to Casper and mark block as received in BlockRetriever") {
+//        Running
+//          .handleBlockMessage[Task](validBlock, sender)(
+//            alwaysDoNotIgnoreF,
+//            alwaysValidBlockF,
+//            casperAddF,
+//            hashHandlerLock
+//          )
+//          .runSyncUnsafe()
+//        sentToCasper.contains(validBlock.blockHash) should be(true)
+//        blockRetriever.isReceived(validBlock.blockHash).unsafeRunSync should be(true)
+//      }
+//
+//      it("should drop not valid messages") {
+//        val b = getRandomBlock().copy(shardId = "wrongshard")
+//        Running
+//          .handleBlockMessage[Task](b, sender)(
+//            alwaysDoNotIgnoreF,
+//            alwaysBadBlockF,
+//            casperAddF,
+//            hashHandlerLock
+//          )
+//          .unsafeRunSync
+//        sentToCasper.contains(b.blockHash) should be(false)
+//      }
+//
+//      it("should not pass message to Casper when should be ignored") {
+//        val b = getRandomBlock().copy(shardId = "wrongshard")
+//        Running
+//          .handleBlockMessage[Task](b, sender)(
+//            alwaysIgnoreF,
+//            alwaysValidBlockF,
+//            casperAddF,
+//            hashHandlerLock
+//          )
+//          .unsafeRunSync
+//        sentToCasper.contains(b.blockHash) should be(false)
+//      }
+//
+//      // TODO this is after LFS comlete
+//      ignore("should drop old ignorable blocks") {}
+//    }
+//  }
```

Comment:
Add reference to an issue where is tracked.

Reply:
https://github.com/rchain/rchain/issues/3277#issuecomment-798109618